scalyr_agent/copying_manager/copying_manager.py

```diff
@@ -312,6 +370,74 @@ def generate_status(self, warn_on_rate_limit=False):
         return result
 
 
+class PathWorkerIdDict(object):
```

Comment:
This is a good abstraction for handling a log file that is managed multiple times!

Reply:
Thanks! I was getting lost without that :)